src/compiler/scala/tools/nsc/transform/AccessorSynthesis.scala

```diff
@@ -332,7 +332,7 @@ trait AccessorSynthesis extends Transform with ast.TreeDSL {
 
         val isUnit    = isUnitGetter(lazyAccessor)
         val selectVar = if (isUnit) UNIT         else Select(thisRef, lazyVar)
-        val storeRes  = if (isUnit) rhsAtSlowDef else Assign(selectVar, rhsAtSlowDef)
+        val storeRes  = if (isUnit) rhsAtSlowDef else Assign(selectVar, fields.castHack(rhsAtSlowDef, lazyVar.info.resultType))
```

Comment:
Removed. I probably wasn't sure if this the symbol was a method (with a `NullaryType` to be stripped with `resultType`) or a field when I wrote that...